esmvaltool/diag_scripts/hydrology/globwat.py

```diff
@@ -0,0 +1,346 @@
+        pet_method_name = 'langbein_'
+    else:
+        pet_method_name = 'debruin_'
+
+    if short_name == 'pet':
+        pet_method = pet_method_name
+    else:
+        pet_method = ''
+
+    base_name = (f"globwat_{dataset_name}_{mip}_{short_name}_{pet_method}"
+                 f"{time_stamp}")
+    filename = get_diagnostic_filename(base_name, cfg, extension=extension)
+    return filename
+
+
+def _shift_era5_time_coordinate(cube):
+    """Shift instantaneous variables 30 minutes forward in time.
+
+    After this shift, as an example:
+    time format [1990, 1, 1, 11, 30, 0] will be [1990, 1, 1, 12, 0, 0].
+    For aggregated variables, already time format is [1990, 1, 1, 12, 0, 0].
+    """
+    if not cube.attributes['mip'] == 'Amon':
+        time = cube.coord(axis='T')
+        time.points = time.points + 30 / (24 * 60)
+        time.bounds = None
+        time.guess_bounds()
+    return cube
+
+
+def main(cfg):
+    """Process data for GlobWat hydrological model.
+
+    These variables are needed in all_vars:
+    pr (precipitation_flux)
+    psl (air_pressure_at_mean_sea_level)
+    rsds (surface_downwelling_shortwave_flux_in_air)
+    rsdt (toa_incoming_shortwave_flux)
+    tas (air_temperature)
+    """
+    # Load target grid to be used in re-gridding
+    target_cube = load_target(cfg)
+
+    input_metadata = cfg['input_data'].values()
+    for dataset_name, metadata in group_metadata(input_metadata,
+                                                 'dataset').items():
+        all_vars, provenance = get_input_cubes(metadata)
+
+        # Fix time coordinate of ERA5 instantaneous variables
+        if dataset_name == 'ERA5':
+            _shift_era5_time_coordinate(all_vars['tas'])
+
+        if cfg['langbein_pet']:
+            logger.info("Calculation PET uisng arora method")
+            all_vars.update(pet=langbein_pet(all_vars['tas']))
+        else:
+            logger.info("Calculation PET uisng debruin method")
+            # Fix time coordinate of ERA5 instantaneous variables
+            if dataset_name == 'ERA5':
+                _shift_era5_time_coordinate(all_vars['psl'])
+            all_vars.update(pet=debruin_pet(
+                psl=all_vars['psl'],
+                rsds=all_vars['rsds'],
+                rsdt=all_vars['rsdt'],
+                tas=all_vars['tas']))
+
+        # Add mip to pet cube attribute
+        all_vars['pet'].attributes['mip'] = all_vars['pr'].attributes['mip']
+        all_vars['pet'].var_name = 'pet'
+
+        # Change negative values for pr to zero
+        _fix_negative_values(all_vars['pr'])
+
+        for key in ['pr', 'pet']:
+            cube = all_vars[key]
+
+            # Convert unit
+            _convert_units(cube)
+
+            # Re-grid data according to the target cube
+            cube = rechunk_and_regrid(cube, target_cube, cfg['regrid_scheme'])
+
+            # Reindex data to a global coordinates set
+            cube = _reindex_data(cube, target_cube)
```

Comment:
I'm not sure I understand why you first regrid, and then have to re-index again, especially since the target cube is the same. I would expect the data after regridding to already be on the same grid as the target_cube. Can you explain?

Reply:
the _reindex_data() is removed now.